nanoFramework.Json/JsonSerializer.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Source code is modified from Mike Jones's JSON Serialization and Deserialization library (https://www.ghielectronics.com/community/codeshare/entry/357)
+
+using System;
+using System.Collections;
+using System.Reflection;
+using System.Text;
+
+namespace nanoFramework.json
+{
+    /// <summary>
+    /// Adapted from JSON.NetMF - JSON Serialization and Deserialization library for .NET Micro Framework
+    /// </summary>
+    public class JsonSerializer
+    {
+        /// <summary>
+        /// Convert an object to a JSON string.
+        /// </summary>
+        /// <param name="o">The value to convert. Supported types are: Boolean, String, Byte, (U)Int16, (U)Int32, Float, Double, Decimal, Array, IDictionary, IEnumerable, Guid, Datetime, TimeSpan, DictionaryEntry, Object and null.</param>
+        /// <returns>The JSON object as a string or null when the value type is not supported.</returns>
+        /// <remarks>For objects, only internal properties with getters are converted.</remarks>
+        internal static string SerializeObject(object o, bool topObject = true)
+        {
+            if (o == null)
+                return "null";
+
+            Type type = o.GetType();
+
+            if (topObject 
+                && !type.IsArray
+                && (type.BaseType.FullName == "System.ValueType"
+                || type.FullName == "System.String"))
+            {
+                return $"[{SerializeObject(o, false)}]";
+            }
+
+            switch (type.Name)
+            {
+                case "Boolean":
+                    {
+                        return (bool)o ? "true" : "false";
+                    }
+                case "TimeSpan":
+                case "String":
+                case "Char":
+                case "Guid":
+                    {
+                        return "\"" + o.ToString() + "\"";
+                    }
+                case "Single":
+                    {
+                        if (float.IsNaN((Single)o))
+                        {
+                            return "null";
+                        }
+                        return o.ToString();
+                    }
+                case "Double":
+                    {
+                        if (double.IsNaN((double)o))
+                        {
+                            return "null";
+                        }
+                        return o.ToString();
+                    }
+                case "Decimal":
+                case "Float":
+                case "Byte":
+                case "SByte":
+                case "Int16":
+                case "UInt16":
+                case "Int32":
+                case "UInt32":
+                case "Int64":
+                case "UInt64":
+                    {
+                        return o.ToString();
+                    }
+                case "DateTime":
+                    { //TODO Remove all reference to non IS8601 DATETime
```

Comment:
please move the comment in the next line and add a space between the // and the TODO